src/java/org/apache/cassandra/db/compaction/LeveledCompactionTask.java

```diff
@@ -56,7 +56,7 @@ public CompactionAwareWriter getCompactionAwareWriter(ColumnFamilyStore cfs,
     @Override
     protected boolean partialCompactionsAcceptable()
     {
-        throw new UnsupportedOperationException("This is now handled in reduceScopeForLimitedSpace");
+        return level == 0;
```

Comment:
`reduceScopeForLimitedSpace` allows to remove sstable on level 0 for task with `level<=1`.

should we change it to `level <= 1`?

Reply:
I think you're right. We could potentially not compact fully expired SSTables if we don't change to `level <= 1` from what I'm reading.

CC @krummas

Reply:
Already fixed in the trunk patch, but I might also add some inline commentary: https://github.com/apache/cassandra/pull/3388

Reply:
Will fix on commit for 4.0, 4.1, and 5.0